pkg/kgo/config.go

```diff
@@ -384,6 +385,13 @@ func (cfg *cfg) validate() error {
 			}
 			cfg.topics[re] = compiled
 		}
+		for re := range cfg.excludeTopics {
+			compiled, err := regexp.Compile(re)
+			if err != nil {
+				return fmt.Errorf("invalid regular expression %q", re)
+			}
+			cfg.excludeTopics[re] = compiled
+		}
 	}
```

Comment:
Mind adding a `else if len(excludeTopics) > 0 { return errors.New("invalid use of ConsumeExcludeTopics when not using ConsumeRegex" }` (or some better error)